### PR TITLE
Changes to support renaming zh-cn to zh-hans

### PR DESCRIPTION
**Pull request by bmedx**
- Added edx_lang_map to config for mapping lang codes to something
different than Transifex has
- Re-enabled skipped tests
- Improved cleanup of test data dirs when tests fail
- Make extract respect config source_locale

### CODE REVIEW

**Comment by bmedx** [comment]
@nedbat would it be possible for you to use this branch the next time you run loc on platform? It shouldn't change anything at this point since platform configuration hasn't been updated to use it, but I'd like to make sure it's not breaking anything before I put those changes in.

**Comment by nedbat** [comment]
@bmedx sure thing.

File: i18n/config.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class Configuration(object):
         'source_locale': 'en',
         'third_party': [],
         'TRANSIFEX_URL': 'https://www.transifex.com/open-edx/edx-platform/',
+        'edx_lang_map': {}
```
Comment (nedbat) [inline i18n/config.py]: Use a trailing comma to make diffs nicer when the next line is eventually added.

File: i18n/extract.py [hunk 2]
```diff
@@ -56,7 +56,10 @@ def rename_source_file(self, src, dst):
         """
         Rename a file in the source directory.
         """
-        os.rename(self.source_msgs_dir.joinpath(src), self.source_msgs_dir.joinpath(dst))
+        try:
+            os.rename(self.source_msgs_dir.joinpath(src), self.source_msgs_dir.joinpath(dst))
+        except OSError:
+            pass
```
Comment (nedbat) [inline i18n/extract.py]: I assume there's a good reason to ignore the error? A comment here will soothe readers' frazzled nerves :)

File: tests/test_generate.py [hunk 3]
```diff
@@ -64,7 +57,7 @@ def test_main(self):
         after start of test suite)
         """
         generate.main(verbose=0, strict=False, root_dir=MOCK_APPLICATION_DIR)
-        for locale in ['mock',]:
+        for locale in ['mock', ]:
```
Comment (nedbat) [inline tests/test_generate.py]: Better is to remove the comma.

File: tests/test_extract.py [hunk 4]
```diff
@@ -1,38 +1,41 @@
...
-# and add this test to it.
-@skip('Long running test')
 class TestExtract(I18nToolTestCase):
     """
     Tests functionality of i18n/extract.py
     """
     generated_files = ('django-partial.po', 'djangojs-partial.po', 'mako.po')
 
-    def setUp(self):
+    def setUp(self, root_dir=MOCK_DJANGO_APP_DIR, preserve_locale_paths=None, clean_paths=None):
```
Comment (nedbat) [inline tests/test_extract.py]: This convention of adding arguments to setUp seems strange to me.  Isn't there a better way to do this?  I haven't seen this done elsewhere.

Comment (bmedx) [reply tests/test_extract.py]: Sure, I can add those elsewhere. I'll clean up that up along with the rest shortly.

File: i18n/extract.py (hunk 2 above)
Comment (bmedx) [reply i18n/extract.py]: The initial reason for this was to help out as I tried to get the files that get left in a changed state after testing under control. Underneath it are some problems with extract getting settings when under test. I've spent too much time trying to get them to work in this task so I'm going to re-skip them in the next commit and add a task to come back and fix that up.

**Comment by nedbat** [comment]
I used commit b3bb328fbf17304bc16b1af8c0ae14aac997f6c8 to collect strings on edx-platform, and it seemed to work fine.

**Comment by bmedx** [comment]
@nedbat if this all looks good to you I can squash this and roll out the new version to PyPI.

**Pull request merged**